datastore/providers/postgres_datastore.py

```diff
@@ -0,0 +1,332 @@
+import os
+from typing import Dict, List, Optional
+from datastore.datastore import DataStore
+from models.models import (
+    DocumentChunk,
+    DocumentChunkMetadata,
+    DocumentMetadataFilter,
+    DocumentChunkWithScore,
+    DocumentMetadataFilter,
+    QueryResult,
+    QueryWithEmbedding,
+)
+import json
+import psycopg
+from psycopg import sql
+from psycopg_pool import ConnectionPool
+import logging
+from services.date import to_unix_timestamp
+
+# Read environment variables for Postgres
+
+POSTGRES_HOST = os.environ.get("POSTGRES_HOST", "localhost")
+POSTGRES_PORT = os.environ.get("POSTGRES_PORT", "5433")
+POSTGRES_USERNAME = os.environ.get("POSTGRES_USERNAME", "postgres")
+POSTGRES_PASSWORD = os.environ.get("POSTGRES_PASSWORD", "")
+POSTGRES_DATABASE = os.environ.get("POSTGRES_DATABASE", "pgml_development")
+POSTGRES_TABLENAME = os.environ.get("POSTGRES_TABLENAME", "chatgpt_datastore")
+POSTGRES_EMBEDDING_INDEX = os.environ.get(
+    "POSTGRES_EMBEDDING_INDEX", "chatgpt_datastore_embedding_index"
+)
+POSTGRES_DOCID_INDEX = os.environ.get(
+    "POSTGRES_EMBEDDING_INDEX", "chatgpt_datastore_docid_index"
```

Comment:
Should this be a different ENV_VAR name than 
```suggestion
    "POSTGRES_DOCID_INDEX", "chatgpt_datastore_docid_index"
```